jsonrpc/base.py

```diff
@@ -51,6 +51,8 @@ class JSONRPCBaseResponse(JSONSerializable):
 
     """ Base class for JSON-RPC 1.0 and JSON-RPC 2.0 responses."""
 
+    request = None  # type: JSONRPCBaseRequest
```

Comment:
Why does "request" attribute belong to class? Should it be implemented on the instance level in __init__ method?

Reply:
fixed